File: LeetCode/Python/#189 Rotate Array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        k = k%len(nums)
        nums[:len(nums)-k],nums[len(nums)-k:] = nums[len(nums)-k:],nums[:len(nums)-k]

# the second way
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        nums.reverse()
        k = k%len(nums)
        nums[:k] = reversed(nums[:k])
        nums[k:] = reversed(nums[k:])

File: LeetCode/JavaScript/#189 Rotate Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */

var rotate = function(nums, k) {
  nums.unshift(...nums.splice(nums.length - k))
};

var rotate = function(nums, k) {
    for(var i=nums.length-1; i>=nums.length-k; i--){
        nums.unshift(nums.pop());
    }
};